code/headers/qik_2s12v10.hpp

```diff
@@ -0,0 +1,132 @@
+
+        /// @brief
+        /// Initializes the qik2s12v10 by resetting it and then setting the baud
+        /// rate.
+        void init();
+        /// @brief
+        /// Sets the brake for the M0 motor.
+        /// @param brake The amount of brake that will be applied to the motor,
+        /// in the range [0,128] where 0 is no brake and 127 is full brake.
+        void set_m0_brake(const uint8_t &brake);
+        /// @brief
+        /// Sets the brake for the M1 motor.
+        /// @param brake The amount of brake that will be applied to the motor,
+        /// in the range [0,128] where 0 is no brake and 127 is full brake.
+        void set_m1_brake(const uint8_t &brake);
+        /// @brief
+        /// Sets the brake for both motors.
+        /// @param brake The amount of brake that will be applied to the motors,
+        /// in the range [0,128] where 0 is no brake and 127 is full brake.
+        void set_brakes(const uint8_t &brake);
+
+        /// @brief
+        /// Returns the errors that the qik2s12v10 has detected since this
+        /// function was last used. The meaning of each bit can be found here:
+        /// https://www.pololu.com/docs/0J29/5.c
+        uint8_t get_error();
+        /// @brief
+        /// Returns the value that the parameter specified by parameter
+        /// currently has.
+        /// @param parameter Specifies the parameter that should be returned.
+        /// The parameters can be found here:
+        /// https://www.pololu.com/docs/0J29/5.a
+        uint8_t get_configuration_parameter(const uint8_t &parameter);
+        /// @brief
+        /// Sets the value of the configurationParameter specified by parameter.
+        /// @param parameter Specifies the parameter that should be set. The
+        /// parameters can be found here: https://www.pololu.com/docs/0J29/5.a
+        /// @param value The value that the configurationParameter should be set
+        /// to.
+        void set_configuration_parameter(const uint8_t &parameter,
+                                         const uint8_t &value);
+
+        /// @brief
+        /// Returns the raw reading from motor M0 that indicates how much
+        /// current flows through the motor at average over the last 5ms. This
+        /// reading is raw, so not converted to milliampere.
+        uint8_t get_m0_current();
+        /// @brief
+        /// Returns the raw reading from motor M1 that indicates how much
+        /// current flows through the motor at average over the last 5ms. This
+        /// reading is raw, so not converted to milliampere.
+        uint8_t get_m1_current();
+        /// @brief
+        /// This function uses the raw reading from motor M0 to estimate how
+        /// much current flows through the motor in milliampere. Note that the
+        /// value returned by this function can differ from the actual current
+        /// by as much as 20%.
+        unsigned int get_m0_current_milliamps();
```

Comment:
Is it not better to return a size_t or a uint16_t for the milliamps? Then we have a fixed size for these variables